crates/fuel-core/src/database/storage.rs

```diff
@@ -1,312 +1,165 @@
-    type OwnedKey = Self::Key;
-    type Value = sparse::Primitive;
-    type OwnedValue = Self::Value;
-}
-
-/// The metadata table for [`ContractsAssetsMerkleData`](ContractsAssetsMerkleData) table
-pub struct ContractsAssetsMerkleMetadata;
-
-impl Mappable for ContractsAssetsMerkleMetadata {
-    type Key = ContractId;
-    type OwnedKey = Self::Key;
-    type Value = SparseMerkleMetadata;
-    type OwnedValue = Self::Value;
-}
-
-/// The table of SMT data for Contract state.
-pub struct ContractsStateMerkleData;
-
-impl Mappable for ContractsStateMerkleData {
-    type Key = [u8; 32];
-    type OwnedKey = Self::Key;
-    type Value = sparse::Primitive;
-    type OwnedValue = Self::Value;
-}
-
-/// The metadata table for [`ContractsStateMerkleData`](ContractsStateMerkleData) table
-pub struct ContractsStateMerkleMetadata;
-
-impl Mappable for ContractsStateMerkleMetadata {
-    type Key = ContractId;
-    type OwnedKey = Self::Key;
-    type Value = SparseMerkleMetadata;
-    type OwnedValue = Self::Value;
-}
-
-/// The table has a corresponding column in the database.
-///
-/// Using this trait allows the configured mappable type to have its'
-/// database integration auto-implemented for single column interactions.
-///
-/// If the mappable type requires access to multiple columns or custom logic during setting/getting
-/// then its' storage interfaces should be manually implemented and this trait should be avoided.
-pub trait DatabaseColumn {
-    /// The column of the table.
-    fn column() -> Column;
-}
-
-impl DatabaseColumn for FuelBlockSecondaryKeyBlockHeights {
-    fn column() -> Column {
-        Column::FuelBlockSecondaryKeyBlockHeights
-    }
-}
-
-impl DatabaseColumn for FuelBlockMerkleData {
-    fn column() -> Column {
-        Column::FuelBlockMerkleData
-    }
-}
-
-impl DatabaseColumn for FuelBlockMerkleMetadata {
-    fn column() -> Column {
-        Column::FuelBlockMerkleMetadata
-    }
-}
-
-impl DatabaseColumn for ContractsAssetsMerkleData {
-    fn column() -> Column {
-        Column::ContractsAssetsMerkleData
-    }
-}
-
-impl DatabaseColumn for ContractsAssetsMerkleMetadata {
-    fn column() -> Column {
-        Column::ContractsAssetsMerkleMetadata
-    }
-}
-
-impl DatabaseColumn for ContractsStateMerkleData {
-    fn column() -> Column {
-        Column::ContractsStateMerkleData
-    }
-}
-
-impl DatabaseColumn for ContractsStateMerkleMetadata {
-    fn column() -> Column {
-        Column::ContractsStateMerkleMetadata
-    }
+use std::borrow::Cow;
+
+/// The trait allows selectively inheriting the implementation of storage traits from `StructuredStorage`
+/// for the `Database`. Not all default implementations of the `StructuredStorage` are suitable
+/// for the `Database`. Sometimes we want to override some of them and add a custom implementation
+/// with additional logic. For example, we want to override the `StorageMutate` trait for the `Messages`
+/// table to also track the owner of messages.
```

Comment:
Would this not apply anymore once we separate executor logic from other kinds of data?

Reply:
Yes, we will remove it after merging https://github.com/FuelLabs/fuel-core/pull/1579 and resolving https://github.com/FuelLabs/fuel-core/issues/1583 and https://github.com/FuelLabs/fuel-core/issues/1580